api/server/index.js

```diff
@@ -101,8 +104,21 @@ const startServer = async () => {
   app.use('/api/roles', routes.roles);
 
   app.use('/api/tags', routes.tags);
+
   app.use((req, res) => {
-    res.sendFile(path.join(app.locals.paths.dist, 'index.html'));
+    // Check for preferred language in cookies, else check accept-language header
+    const preferredLanguage =
+      req.cookies.langcode || req.headers['accept-language']?.split(',')[0] || 'en-US';
+    const indexPath = path.join(app.locals.paths.dist, 'index.html');
+
+    fs.readFile(indexPath, 'utf8', (err, data) => {
```

Comment:
I don't think we should read file on every request here, maybe only read it once at server startup?

`res.sendFile()` is efficient and this change is not

Reply:
Good catch, this should now be fixed.